packages/glimmer-runtime/lib/syntax/core.ts

```diff
@@ -924,6 +926,38 @@ export class Helper extends ExpressionSyntax<Opaque> {
   }
 }
 
+export class HasBlock extends ExpressionSyntax<boolean> {
+  type = "has-block";
+
+  static fromSpec(sexp: SerializedExpressions.HasBlock): HasBlock {
+    let [, name] = sexp;
+
+    return new HasBlock({
+      name: name as InternedString
+    });
+  }
+
+  static build(name: InternedString): HasBlock {
+    return new this({ name });
+  }
+
+  name: InternedString;
+
+  constructor({ name }: { name: InternedString }) {
+    super();
+    this.name = name;
+  }
+
+  prettyPrint() {
+    return new PrettyPrint('expr', 'has-block', [this.name]);
+  }
```

Comment:
I don't think we need `prettyPrint` anymore (but maybe it's needed to avoid a dangling type error). We do, however, need a `toJSON`.